src/main/scala/io/iohk/ethereum/network/p2p/validators/BlockValidator.scala

```diff
@@ -0,0 +1,107 @@
+      block.body.transactionList,
+      new RLPByteArraySerializable[SignedTransaction]
+    )
+    if (isValid) Right(block)
+    else Left(BlockTransactionsHashError)
+  }
+
+  /**
+    * Validates [[BlockBody.uncleNodesList]] against [[io.iohk.ethereum.domain.BlockHeader.ommersHash]]
+    * based on validations stated in section 4.2.2 of http://paper.gavwood.com/
+    *
+    * @param block Block to validate
+    * @return Block if valid, a Some otherwise
+    */
+  private def validateOmmers(block: Block): Either[BlockError, Block] = {
+    // FIXME Can we avoid encoding ommers again?
+    val encodedOmmers = encode(BlockBody.rlpEncDec.encodeUncles(block.body.uncleNodesList))
+    if (sha3(encodedOmmers) sameElements block.header.ommersHash) Right(block)
+    else Left(BlockOmmersHashError)
+  }
+
+  /**
+    * Validates [[Receipt]] against [[io.iohk.ethereum.domain.BlockHeader.receiptsRoot]]
+    * based on validations stated in section 4.2.2 of http://paper.gavwood.com/
+    *
+    * @param block    Block to validate
+    * @param receipts Receipts to use
+    * @return
+    */
+  private def validateReceipts(block: Block, receipts: Seq[Receipt]): Either[BlockError, Block] = {
+
+    val isValid = MptListValidator.isValid[Receipt](block.header.receiptsRoot.toArray[Byte],
+      receipts,
+      new RLPByteArraySerializable[Receipt]
+    )
+    if (isValid) Right(block)
+    else Left(BlockReceiptsHashError)
+  }
+
+  /**
+    * Validates [[io.iohk.ethereum.domain.BlockHeader.logsBloom]] against [[Receipt.logsBloomFilter]]
+    * based on validations stated in section 4.2.2 of http://paper.gavwood.com/
+    *
+    * @param block    Block to validate
+    * @param receipts Receipts to use
+    * @return
+    */
+  private def validateLogBloom(block: Block, receipts: Seq[Receipt]): Either[BlockError, Block] = {
+    val logsBloomOr = receipts.foldLeft(Array.fill(block.header.logsBloom.size)(0.toByte)) { (or: Array[Byte], receipt: Receipt) =>
+      or.zip(receipt.logsBloomFilter.toArray[Byte]).map(b => (b._1 | b._2).toByte)
+    }
+    if (logsBloomOr sameElements block.header.logsBloom.toArray[Byte]) Right(block)
+    else Left(BlockLogBloomError)
+  }
+
+  /**
+    * This method allows validate a Block. It only perfoms the following validations (stated on
+    * section 4.2.2 of http://paper.gavwood.com/):
+    *   - [[BlockValidator.validateTransactionRoot]]
+    *   - [[BlockValidator.validateOmmers]]
+    *   - [[BlockValidator.validateReceipts]]
+    *   - [[BlockValidator.validateLogBloom]]
+    *
+    * @param block  Block to validate
+    * @param receipts Receipts to be in validation process
+    * @return The block if validations are ok, error otherwise
+    */
+  def validate(block: Block, receipts: Seq[Receipt]): Either[BlockError, Block] = {
+    for {
+      _ <- validateTransactionRoot(block)
+      _ <- validateOmmers(block)
+      _ <- validateReceipts(block, receipts)
+      _ <- validateLogBloom(block, receipts)
```

Comment:
I'm not sure if this answers your question but a simple proof of work validation is already implemented in `validatePoW` from `BlockHeaderValidation` (there's also a task to complete this validation https://iohk.myjetbrains.com/youtrack/issue/EC-88).